### PR TITLE
[wasm][debugger] Fix firefox tests

### PR DESCRIPTION
**Pull request by thaystg**
The tests were broken since this PR: https://github.com/dotnet/runtime/pull/72812
Unfortunatelly I forgot to click on the firefox lane, I believed in the green check.

### CODE REVIEW

File: src/mono/wasm/debugger/BrowserDebugProxy/Firefox/FirefoxMonoProxy.cs [hunk 1]
```diff
@@ -794,6 +794,19 @@ private static JObject ConvertToFirefoxContent(ValueOrError<GetMembersResult> re
...
 
+    protected override async Task<bool> IsRuntimeAlreadyReadyAlready(SessionId sessionId, CancellationToken token)
+    {
+        if (contexts.TryGetValue(sessionId, out ExecutionContext context) && context.IsRuntimeReady)
+            return true;
+
+        Result res = await SendMonoCommand(sessionId, MonoCommands.IsRuntimeReady(RuntimeId), token);
+        if (!res.IsOk) //if runtime is not ready this may be the response
+            return false;
+        if (res.Value?["result"]?["value"] is JObject) //if runtime is not ready this may be the response
```
Comment (radical) [inline src/mono/wasm/debugger/BrowserDebugProxy/Firefox/FirefoxMonoProxy.cs:805]: Should we instead check for `res.Value?["result"]?["value"]?.Type == JTokenType.Boolean`?

Comment (radical) [reply src/mono/wasm/debugger/BrowserDebugProxy/Firefox/FirefoxMonoProxy.cs:805]: Can we do that in the base method itself?

**Comment by ilonatommy** [comment]
@radical, does failing `runtime-wasm-non-libtests` not fail the whole run on purpose? We used to have problems with `staging` that remained green even when having failures, can we change `non-libtest` behavior it to be more visible?

**Comment by radical** [comment]
> > @radical, does failing `runtime-wasm-non-libtests` not fail the whole run on purpose? We used to have problems with `staging` that remained green even when having failures, can we change `non-libtest` behavior it to be more visible?
> 
> For Firefox tests that's a deliberate choice right now. But if we feel that the Firefox tests are stable enough (and we can watch them for a week or two) then we can change the behavior.

Just to add more context there - it used to be pipeline dependent earlier, but now we control this on a per-job basis.

File: src/mono/wasm/debugger/BrowserDebugProxy/Firefox/FirefoxMonoProxy.cs (hunk 1 above)
Comment (thaystg) [reply src/mono/wasm/debugger/BrowserDebugProxy/Firefox/FirefoxMonoProxy.cs]: done

**Review by radical — APPROVED** [review]

**Review by ilonatommy — APPROVED** [review]

**Pull request merged**